sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java

```diff
@@ -328,9 +332,22 @@ private void logMessageWithFormat(String levelName, String format, Object... arg
     private void log(String levelName, String message, Throwable t) {
         String dateTime = getFormattedDate();
         String threadName = Thread.currentThread().getName();
-        StringBuilder buf = new StringBuilder(64);
-        buf.append(String.format(MESSAGE_TEMPLATE, dateTime, threadName, levelName, classPath, message));
-        writeWithThrowable(buf, t);
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
if the total length is known you can pass that to the constructor of StringBuilder to avoid re-growing cost.